File: Quasics2021Code/MaeTester2022/src/main/cpp/commands/RunOnlyIntakeMotorReverse.cpp
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

#include "commands/RunOnlyIntakeMotorReverse.h"

RunOnlyIntakeMotorReverse::RunOnlyIntakeMotorReverse(Intake* intake)
    : intake(intake) {
  // Use addRequirements() here to declare subsystem dependencies.
  AddRequirements(intake);
}

// Called when the command is initially scheduled.
void RunOnlyIntakeMotorReverse::Initialize() {
  intake->OnlyIntakeReverse();
}

// Called once the command ends or is interrupted.
void RunOnlyIntakeMotorReverse::End(bool interrupted) {
  intake->OnlyIntakeOff();
}

File: Quasics2026Code/MaeTester2026-Java/src/main/java/frc/robot/commands/RunOnlyIntakeMotorReverse.java
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

package frc.robot.commands;

import edu.wpi.first.wpilibj2.command.Command;
import frc.robot.subsystems.Intake;

public class RunOnlyIntakeMotorReverse extends Command {
  private final Intake m_intake;
  /** Creates a new RunOnlyIntakeMotorReverse. */
  public RunOnlyIntakeMotorReverse(Intake intake) {
    m_intake = intake;
    // Use addRequirements() here to declare subsystem dependencies.
    addRequirements(intake);
  }

  // Called when the command is initially scheduled.
  @Override
  public void initialize() {
    m_intake.onlyIntakeReverse();
  }

  // Called once the command ends or is interrupted.
  @Override
  public void end(boolean interrupted) {
    m_intake.onlyIntakeOff();
  }
}
